### ISSUE TITLE
Rig metadata conditioning does not match the paper: wrong mechanism, wrong representation, 3 of 4 fields missing

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Rig3R#37]
## Problem

The paper's central contribution is rig-aware conditioning. The current implementation
conditions on rig metadata through a mechanism that differs from the paper in three ways
at once, all stemming from the same design choice: metadata was modelled as *an SE(3)
side-channel token* rather than *a raymap added to every patch*.

Verified against Rig3R (arXiv 2506.02265) §3.2 and §3.3.

### 1. Metadata is prepended as separate tokens, not added to patch tokens

§3.3, Metadata Embedding:

> All components are concatenated and **added to the patch tokens**.

`models/decoder_transformer.py` concatenates metadata as extra *sequence* entries in
front of the patch tokens, then slices them back off after the transformer layers:

```python
seq = torch.cat([meta_tokens, tokens], dim=1)          # line 175
for layer in self.layers:
    seq = layer(seq)
proc_patches = seq[:, meta_tokens.shape[1]:, :]        # line 182
```

Additive per-patch conditioning and prepended tokens are different architectures, not a
tuning difference. In the paper every patch carries its own metadata; here the metadata
reaches patches only indirectly through attention, and is discarded before the heads.

### 2. Rig metadata is a flattened 4x4 SE(3), not a per-patch 6-channel raymap

§3.2 defines the rig pose field as `r_i ∈ R^(H×W×6)`, and §3.3 says the **rig raymap
patch** `r_i ∈ R^6` is linearly projected to the model dimension. It is a spatially
aligned, per-patch signal — that alignment is the stated reason raymaps are preferred
over pose matrices (§3.1).

`models/decoder_transformer.py:78`:

```python
self.key_projs = nn.ModuleDict({
    "cam2rig": nn.Linear(16, embed_dim)  # one token per view, a flattened 4x4 SE(3)
})
```

One token per view, from a 4x4 matrix. Table 3 attributes the entire unseen-rig
generalization gain to this field (WayveScenes101 mAA 25.7 → 56.4), so the
representation is not incidental.

### 3. Three of the four metadata fields do not exist

§3.3 lists frame index `N`, camera ID `c_i`, timestamp `t_i`, and rig raymap `r_i`.
`N` and `c_i` use 1D sine-cosine embeddings over a randomly sampled larger index range;
`t_i` is normalized seconds, encoded the same way.

Only `cam2rig` is implemented. Notably the frame index is the field the paper says is
*always* included, "to uniquely identify each image within the transformer" — without it
the decoder runs joint self-attention over all views with nothing distinguishing one view
from another.

`datasets/co3d.py:135-138` already has dropout branches for `camera_id` and `timestamp`,
but `_load_metadata` never produces those keys, so the branches are unreachable.

## Suggested fix

1. Build the per-patch rig raymap as metadata input. `utils/raymap.py` already computes
   exactly this tensor for the *target* side (`build_raymap_targets` -> `rig_raymap`,
   shape `(B, V, P, 6)`); the input path can reuse it.
2. Replace the prepend-and-slice path with additive conditioning: project each metadata
   component to `embed_dim`, sum them, add to the patch tokens before the decoder layers.
3. Add frame index, camera ID, and timestamp with 1D sine-cosine embeddings. Emit them
   from the datasets (the CO3D dropout branches already anticipate the key names).

Note that `models/rig3r.py:61` currently reads `metadata["cam2rig"]` guarded only by
`if metadata`, so adding any other metadata key without `cam2rig` present will raise a
`KeyError`. Worth fixing as part of this.

## Deliberate divergences worth documenting, not fixing

These are compute-budget choices, listed so they are not mistaken for bugs later:

- **Decoder depth.** §3.3 specifies "a second ViT-Large transformer, trained from
  scratch" (24 layers, 16 heads). We run 2 layers, 8 heads. The paper used 128 H100s for
  5 days; this repo targets a single 12 GB card.
- **Image resolution.** Paper resizes to 512x512 with padding; we use 128x128.

Both should be stated in the README so reported numbers are not read as reproductions.

Related: #34 (encoder was not actually loading DUSt3R weights).

**Comment by engichang1467** [comment]
Paper-alignment audit against Rig3R (arXiv 2506.02265), filed as a set:

- #37 — metadata conditioning: mechanism, representation, missing fields (this issue)
- #38 — `RigRaymapHead` leaks ground-truth `cam2rig`
- #39 — metadata dropout substitutes identity instead of dropping
- #40 — raymap heads missing camera-center MLP; dead loss branches
- #41 — losses diverge from Eq. 3 / Eq. 4
- #42 — encoder has no positional encoding (tail of #34)

Cross-references in the bodies above point at these by description; numbers are here.

Suggested order: #42 and #40 are self-contained and unblock supervision that currently
does nothing. #38 and #39 are small and make training honest. #37 is the large one and
#41 will need loss-weight retuning after.
